commit 45cf86f26148e549c5ba4a8ab32a390e4bde216e
Author: Emanuele Giuseppe Esposito <eesposit@redhat.com>
Date:   Wed Feb 1 08:29:05 2023 -0500

    kvm: x86: Advertise FLUSH_L1D to user space
    
    FLUSH_L1D was already added in 11e34e64e4103, but the feature is not
    visible to userspace yet.
    
    The bit definition:
    CPUID.(EAX=7,ECX=0):EDX[bit 28]
    
    If the feature is supported by the host, kvm should support it too so
    that userspace can choose whether to expose it to the guest or not.
    One disadvantage of not exposing it is that the guest will report
    a non existing vulnerability in
    /sys/devices/system/cpu/vulnerabilities/mmio_stale_data
    because the mitigation is present only if the guest supports
    (FLUSH_L1D and MD_CLEAR) or FB_CLEAR.
    
    Signed-off-by: Emanuele Giuseppe Esposito <eesposit@redhat.com>
    Message-Id: <20230201132905.549148-4-eesposit@redhat.com>
    Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>

diff --git a/arch/x86/kvm/cpuid.c b/arch/x86/kvm/cpuid.c
index 599aebec2d52..9583a110cf5f 100644
--- a/arch/x86/kvm/cpuid.c
+++ b/arch/x86/kvm/cpuid.c
@@ -653,7 +653,7 @@ void kvm_set_cpu_caps(void)
 		F(SPEC_CTRL_SSBD) | F(ARCH_CAPABILITIES) | F(INTEL_STIBP) |
 		F(MD_CLEAR) | F(AVX512_VP2INTERSECT) | F(FSRM) |
 		F(SERIALIZE) | F(TSXLDTRK) | F(AVX512_FP16) |
-		F(AMX_TILE) | F(AMX_INT8) | F(AMX_BF16)
+		F(AMX_TILE) | F(AMX_INT8) | F(AMX_BF16) | F(FLUSH_L1D)
 	);
 
 	/* TSC_ADJUST and ARCH_CAPABILITIES are emulated in software. */